my/core/__main__.py

```diff
@@ -360,69 +329,121 @@ def module_install(args: Namespace) -> None:
     check_call(cmd)
 
 
-from argparse import ArgumentParser
-def parser() -> ArgumentParser:
-    p = ArgumentParser('Human Programming Interface', epilog='''
-Tool for HPI.
+@click.group()
+def main() -> None:
+    '''
+    Human Programming Interface
 
-Work in progress, will be used for config management, troubleshooting & introspection
-''')
-    sp = p.add_subparsers(dest='mode')
-    dp = sp.add_parser('doctor', help='Run various checks')
-    dp.add_argument('--verbose', action='store_true', help='Print more diagnosic infomration')
-    dp.add_argument('--all'    , action='store_true', help='List all modules, including disabled')
-    dp.add_argument('--quick'  , action='store_true', help='Only run partial checks (first 100 items)')
-    dp.add_argument('module', nargs='?', type=str   , help='Pass to check a specific module')
-    dp.set_defaults(func=doctor)
+    Tool for HPI
+    Work in progress, will be used for config management, troubleshooting & introspection
+    '''
+    # note: code in groups doesnt run unless a subcommand is passed
```

Comment:
Hmm, what does it imply? As in, isn't that what one would expect? 
As far as I can tell from the docs it basically means it lets to get rid of things like `print_help` handling (nice!)? And I guess often there is a pattern in argparse where you do `if/elif` switching on `args.subcommand` or something like that, which also isn't a problem in click?

Reply:
Ah, yeah -- previously I was using `tempfile.mkdtemp` which ended up creating a bunch of temporary directories in `/tmp`, so it was sort of a warning. I just left it there and added the links incase you wanted to add some global flags/context later on

Reply:
Tangentially related, you can choose to run a group if a subcommand isnt passed:

```
@click.group(invoke_without_command=True)
@click.pass_context
def cli(ctx):
    if ctx.invoked_subcommand is None:
        click.echo('I was invoked without subcommand')
    else:
        click.echo('I am about to invoke %s' % ctx.invoked_subcommand)

@cli.command()
def sync():
    click.echo('The subcommand')
```

If a subcommand is passed, it runs the function body, then invokes the subcommand